sktime/markov/msm/augmented_msm.py

```diff
@@ -0,0 +1,623 @@
+            raise ValueError("Experimental measurements must span full observable state space.")
+
+        count_matrix = count_model.count_matrix
+        if issparse(count_matrix):
+            count_matrix = count_matrix.toarray()
+
+        # slice out active states from E matrix
+        expectations_selected = self.expectations_by_state[count_model.state_symbols]
+        count_matrix_symmetric = 0.5 * (count_matrix + count_matrix.T)
+        nonzero_counts = np.nonzero(count_matrix_symmetric)
+        counts_row_sums = np.sum(count_matrix, axis=1)
+        expectations_confidence_interval = confidence_interval(expectations_selected, conf=self.support_confidence)
+
+        measurements = self.experimental_measurements
+        measurement_weights = self.experimental_measurement_weights
+
+        count_outside = []
+        count_inside = []
+
+        i = 0
+        # Determine which experimental values are outside the support as defined by the Confidence interval
+        for emi, ema, mm, mw in zip(expectations_confidence_interval[0], expectations_confidence_interval[1],
+                                    measurements, measurement_weights):
+            if mm < emi or ema < mm:
+                self._log.info("Experimental value %f is outside the support (%f,%f)" % (mm, emi, ema))
+                count_outside.append(i)
+            else:
+                count_inside.append(i)
+            i = i + 1
+
+        # A number of initializations
+        transition_matrix, stationary_distribution = msmest.tmatrix(count_matrix, reversible=True,
+                                                                    return_statdist=True)
+        if issparse(transition_matrix):
+            transition_matrix = transition_matrix.toarray()
+        # Determine number of slices of R-tensors computable at once with the given cache size
+        slices_z = np.floor(self.max_cache / (transition_matrix.nbytes / 1.e6)).astype(int)
+        # Optimizer state
+        state = AMMOptimizerState(expectations_selected, measurements, measurement_weights,
+                                  stationary_distribution, slices_z, count_matrix_symmetric, counts_row_sums)
+        ll_old = state.log_likelihood_biased(count_matrix, transition_matrix)
+
+        state.log_likelihoods.append(ll_old)
+        # make sure everything is initialized
+        state.update_pi_hat()
+        state.update_m_hat()
+        state.update_Q()
+        state.update_X_and_pi()
+
+        ll_old = state.log_likelihood_biased(count_matrix, transition_matrix)
+        state.log_likelihood_prev = ll_old
+        state.update_G()
+
+        #
+        # Main estimation algorithm
+        # 2-step algorithm, lagrange multipliers and pihat have different convergence criteria
+        # when the lagrange multipliers have converged, pihat is updated until the log-likelihood has converged
+        # (changes are smaller than 1e-3).
```

Comment:
again is there a reason for `1e-3` to not be tunable

Reply:
maybe, wasn't tunable in the original code. An empirical value probably